crates/iota-sdk/src/iota_client_config.rs

```diff
@@ -99,12 +100,11 @@ impl IotaClientConfig {
 
     /// Add an [`IotaEnv`].
     pub fn add_env(&mut self, env: IotaEnv) {
-        if !self
-            .envs
-            .iter()
-            .any(|other_env| other_env.alias == env.alias)
-        {
-            self.envs.push(env)
+        if self.get_env(&env.alias).is_none() {
+            if self.active_env.is_none() {
+                self.set_active_env(env.alias.clone());
+            }
+            self.envs.push(env);
```

Comment:
I think the environment should also be set to active n case the user tried to add the same environment again.

Reply:
I'm not sure I understand. You mean if the env exists and they try to add it again?

Reply:
yes

Reply:
I don't think I agree